components/bind/Solaris/dns-server.sh

```diff
@@ -20,34 +20,39 @@
 # CDDL HEADER END
 #
 # Copyright (c) 2007, 2012, Oracle and/or its affiliates. All rights reserved.
+# Copyright 2016 Hans Rosenfeld <rosenfeld@grumpf.hope-2000.org>
 #
-
 # smf_method(5) start/stop script required for server DNS
 
 . /lib/svc/share/smf_include.sh
 
-result=${SMF_EXIT_OK}
-
-# Read command line arguments
-method="$1"		# %m
-instance="$2" 		# %i
+mount_chroot ()
+{
+    c=$1
+    shift
+    for f in $*; do
+        if [ -z "${f}" -o ! -f "${f}" -o \
+             -z "${c}" -o ! -d "${c}" ]; then
+             exit ${SMF_EXIT_ERR_CONFIG}
+        fi
 
-# Set defaults; SMF_FMRI should have been set, but just in case.
-if [ -z "$SMF_FMRI" ]; then
-    SMF_FMRI="svc:/network/dns/server:${instance}"
-fi
-server="/usr/sbin/named"
-I=`/usr/bin/basename $0`
+        umount ${c}/${f}
```

Comment:
This is just to make sure it actually is not mounted already before mounting it. Under normal operation that is noop, so I chose to supress error messages.